### ISSUE TITLE
Cannot interact with a smartcontract using a private blockscout instance

### ISSUE THREAD

**Issue by adrienthiery** [blockscout/blockscout#5803]
Hi there,

I deployed a blockscout instance to explore a private (ganache) blockchain for development, and all works pretty well except when we want to interact with the contracts deployed from the "Write Contract" tab, we get the following error : 
![image](https://user-images.githubusercontent.com/2392118/183051491-a2ef5ee7-f57c-4ad0-845f-0bf5bbbbf465.png)

```
Unauthorized
You connected to Unknown Private Network chain in the wallet, but the current instance of Blockscout is for Unknown Private Network chain
```

Followed by this error: 
![image](https://user-images.githubusercontent.com/2392118/183051656-f4ef09d6-16b1-4519-b911-e77846645c68.png)

```
No "from" address specified in neither the given options, nor the default options.
```

I don't see anything in the logs of the container

I tried to find a mention of this in the documentation, stackoverflow and issues, but could not find anything.

This is probably a setup error/missing step, but I can't find anything to resolve this.

### Environment

* Elixir & Erlang/OTP versions (`elixir -version`): 1.13.1
* Operating System: Docker linux (using https://hub.docker.com/layers/alpine-elixir-phoenix/bitwalker/alpine-elixir-phoenix/1.13/images/sha256-ab38ca0743abba623298414e9b0f0cddd64e73606b04a86d72a9eaeac1a26aa6 as a base image)
* Blockscout Version/branch: master from July 13th (i.e. https://github.com/blockscout/blockscout/commit/ef43da6d42b5aae00960fe42e6bc3cabc78d80d2 )

### Steps to reproduce

*Tell us how to reproduce this issue.  ❤️ if you can push up a branch to your fork with a regression test we can run to reproduce locally.*

### Expected behaviour

I should be able to interact with my smartcontract on a blockscout instance

### Actual behaviour

Getting an error when trying to.

**Comment by nikitosing** [comment]
@adrienthiery how did you connect metamask to your ganache blockchain?

**Comment by adrienthiery** [comment]
Hi @nikitosing ,
I added a network with the RPC URL of my ganache blockchain to metamask, then connected by clicking the "connect wallet" button on the `Read contract` page of my blockscout instance.

**Comment by adrienthiery** [comment]
Hello @nikitosing, any idea how to help me out here? 🤔

**Comment by nikitosing** [comment]
@adrienthiery Please check did you set correct `CHAIN_ID` env variable? Then check if metamask connected to correct network

**Comment by vbaranov** [comment]
I suppose the issue is specific to Docker setup. Specifically, how we forwarding backend env vars to front. This PR #6017 is aimed to resolve this issue.

**Comment by vbaranov** [comment]
@adrienthiery PR merged. You are good to go with the latest master branch - please pull updates from there. And here is also slightly modification to your Ganache run: you should fixate networkID by running `ganache --networkId 1337`

**Comment by adrienthiery** [comment]
Nice! I will try it now. Thanks @vbaranov

**Comment by adrienthiery** [comment]
Mmmh, sadly I seem to still get the same error : The same two pop-ups are coming up.

Chrome's console says:
```
Uncaught (in promise) Error: No "from" address specified in neither the given options, nor the default options.
    at Object.ContractNoFromAddressDefinedError (smart-contract-helpers.js:2:2192881)
    at u._executeMethod (smart-contract-helpers.js:2:2307614)
    at smart-contract-helpers.js:2:3697352
```

I did verify that the env vars set for `NETWORK_ID` and `CHAIN_ID` use the good values 🤔

**Comment by adrienthiery** [comment]
What's very strange is that it is not really a connection issue : 
`Connect Wallet` works fine, it does pop my Metamask up and asks to connect.
Also, I can read data from the `READ CONTRACT` tab with no issue, it's really the `WRITE` action that poses problem with this `No "from" address` error

**Comment by adrienthiery** [comment]
[Strange thing here](https://github.com/blockscout/blockscout/blob/5de55bcbde703e43536af54c9222b7b448ccc0a4/apps/block_scout_web/lib/block_scout_web/templates/smart_contract/_functions.html.eex#L69), when inspecting the form, I see that `data-chain-id` is actually set to the `networkId` and not the `chainId`, which in our case are indeed different...

**Comment by adrienthiery** [comment]
I did not restart ganache, as we use it in a shared deployed environment (it's not on my local machine), so I can't reset it right now.

When running the curl command, I do get `{"id":83,"jsonrpc":"2.0","result":"1657732650246"}` back from ganache.
(Which is the `networkID` we use for that test blockchain)

But we run ganache with : `ganache --chainId 8888` (I will add the `--networkId` next time I can reset the blockchain, but I can't right now).

This whole `chainId` vs `networkId` is very confusing.
Metamask asks for the `chainId` to be setup when adding the network, but the `networkId` seems to be something blurrier.

I know that for most chain, they match

**Comment by adrienthiery** [comment]
Should we not use the `data-chain-id="<% env.CHAIN_ID %>` instead of `data-chain-id="<%= Explorer.Chain.Cache.NetVersion.get_version() %>"` in here: https://github.com/blockscout/blockscout/blob/5de55bcbde703e43536af54c9222b7b448ccc0a4/apps/block_scout_web/lib/block_scout_web/templates/smart_contract/_functions.html.eex#L69 ?

**Comment by vbaranov** [comment]
> Should we not use the data-chain-id="<% env.CHAIN_ID %> instead of data-chain-id="<%= Explorer.Chain.Cache.NetVersion.get_version() %>" in here:

Yes, probably, good idea. Let me apply.

**Comment by vbaranov** [comment]
@adrienthiery can you try pulling update from master branch? Don't forget to change `CHAIN_ID` witch matching actual chain ID of your Ganache run.

**Comment by adrienthiery** [comment]
Yay! That works! Thanks a lot @vbaranov :)

**Comment by jensiepoo** [comment]
Sorry, why was this reverted right after? It's breaking again for me. https://github.com/blockscout/blockscout/commit/41709f6d734674ff01d1895e7a79f8172b55a7cc#diff-d1676759ea023d079059d8bcf6483b52d9b6f88040627fafa20efb0416f74b58

**Comment by nikitosing** [comment]
@jensiepoo Did you set `CHAIN_ID`? And what the error?

**Comment by jensiepoo** [comment]
Same error as @adrienthiery. Would you be able to provide more context as to why 
`data-chain-id="<%= Application.get_env(:block_scout_web, :chain_id) %>"` got reverted back to `data-chain-id="<%= Explorer.Chain.Cache.NetVersion.get_version() %>"`

**Comment by jensiepoo** [comment]
For more context, the html element based on that template came out to be

`<form class="form-inline" data-function-form="" data-action="write" data-type="regular" data-url="/smart-contracts/0x8CdaF0CD259887258Bc13a92C0a6dA92698644C0" data-contract-address="0x8cdaf0cd259887258bc13a92c0a6da92698644c0" data-contract-abi="[{&quot;stateMutability&quot;:&quot;payable&quot;,&quot;type&quot;:&quot;fallback&quot;}]" data-implementation-abi="[{&quot;stateMutability&quot;:&quot;payable&quot;,&quot;type&quot;:&quot;fallback&quot;}]" data-chain-id="1674042004529" data-custom-abi="false">
          <input type="hidden" name="function_name" value="">
          <input type="hidden" name="method_id" value="">


            <div class="form-group pr-3 d-flex">
              <input type="number" name="function_input" tx-value="" data-toggle="tooltip" title="" class="form-control form-control-sm address-input-sm" placeholder="value(ETH)" min="0" step="1e-18" data-original-title="Amount in native token <ETH>">
            </div>

          <div>
            <input type="submit" value="Write" class="button btn-line button-xs py-0 write-contract-btn">
          </div>
        </form>`

Where the `data-chain-id` would've been correct if we used `"<%= Application.get_env(:block_scout_web, :chain_id) %>"` instead.

**Comment by tina1998612** [comment]
I'm still facing this error, is it fixed now?

**Comment by nikitosing** [comment]
yes, it should be fixed

**Comment by ShinariumChain** [comment]
Not fixed in v5.2.3-beta?

**Comment by orshih6** [comment]
i'm facing this error. does it fixed ?
